content/webapp/components/EventSchedule/EventScheduleItem.tsx

```diff
@@ -98,30 +99,58 @@ const eventLocations = (locations: Place[], isHybridEvent: boolean) => {
 
 // We have a message block on scheduled events which either displays
 // 'Just turn up' or 'Arrive early to register'
-// We only show this message if:
-// - the event isn't past AND
-// - it doesn't require booking (either through Eventbrite or by contacting the booking enquiry team) AND
-// - it doesn't have it's own event schedule AND
-// - the parent event isn't Ticketed
-function shouldShowMessage({
-  event,
-  parentEvent,
-}: {
-  event: Event;
-  parentEvent: Event;
-}): boolean {
+// N.B. If the criteria to display both messages is satisfied, we only show 'Arrive early to register' (see below)
+// We don't show either message if the following criteria aren't satisfied:
+function shouldShowMessage(event: Event): boolean {
   return (
     !isEventPast(event) &&
     !event.eventbriteId &&
     !event.bookingEnquiryTeam &&
-    !(event.schedule && event.schedule.length > 1) &&
-    parentEvent.bookingType !== 'Ticketed'
+    !(event.schedule && event.schedule.length > 1)
+  );
+}
+// We should show the 'Arrive early to register' message if:
+// - event.hasEarlyRegistration is true AND
+// - the event isn't past AND
+// - it doesn't require booking (either through Eventbrite or by contacting the booking enquiry team) AND
+// - it doesn't have it's own event schedule
+function shouldShowEarlyRegistrationMessage(event: Event): boolean {
+  return event.hasEarlyRegistration;
+}
+
+// We should show the 'Just turn up' message if 'drop in at any time' policy is added to the event AND
+function shouldShowJustTurnUpMessage(event: Event): boolean {
+  const hasDropInPolicy = event.policies.some(
+    p => p.id === eventPolicyIds.dropIn
   );
+  if (hasDropInPolicy) {
+    return true;
+  } else {
+    return false;
+  }
```

Comment:
np; Could it be `return hasDropInPolicy` ? Obviously very readable as is as well.